src/code_1A5830.c

```diff
@@ -18,19 +18,77 @@ INCLUDE_ASM(s32, "code_1A5830", dispatch_damage_event_actor_0);
 
 INCLUDE_ASM(s32, "code_1A5830", dispatch_damage_event_actor_1);
 
-INCLUDE_ASM(s32, "code_1A5830", BindTakeTurn);
-
-INCLUDE_ASM(s32, "code_1A5830", PauseTakeTurn);
-
-INCLUDE_ASM(s32, "code_1A5830", ResumeTakeTurn);
+ApiStatus BindTakeTurn(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorId actorID = get_variable(script, *args++);
+    s32 var1;
+
+    if (actorID == ActorId_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    var1 = get_variable(script, *args++);
+    get_actor(actorID)->takeTurnCode = var1;
+    return ApiStatus_DONE2;
+}
+
+ApiStatus PauseTakeTurn(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorId actorID = get_variable(script, *args++);
+
+    if (actorID == ActorId_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    get_variable(script, *args++);
+    suspend_all_script(get_actor(actorID)->takeTurnID);
+    return ApiStatus_DONE2;
+}
+
+ApiStatus ResumeTakeTurn(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorId actorID = get_variable(script, *args++);
+
+    if (actorID == ActorId_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    get_variable(script, *args++);
+    resume_all_script(get_actor(actorID)->takeTurnID);
+    return ApiStatus_DONE2;
+}
 
 INCLUDE_ASM(s32, "code_1A5830", BindIdle);
 
 INCLUDE_ASM(s32, "code_1A5830", EnableIdleScript);
 
-INCLUDE_ASM(s32, "code_1A5830", BindHandleEvent);
+ApiStatus BindHandleEvent(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    ActorId actorID = get_variable(script, *args++);
+    s32 var1;
+
+    if (actorID == ActorId_SELF) {
+        actorID = script->owner1.actorID;
+    }
+
+    var1 = get_variable(script, *args++);
+    get_actor(actorID)->onHitCode = var1;
```

Comment:
if we're really going full out, assuming it matches, we might even be able to do:
`get_actor(actorID)->onHitCode = get_variable(script, *args++);`

Reply:
I tried just to see earlier, but we need the temp for matching